internal/log/slog_test.go

```diff
@@ -0,0 +1,175 @@
+// Copyright The prometheus-operator Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package log
+
+import (
+	"bytes"
+	"encoding/json"
+	"log/slog"
+	"os"
+	"reflect"
+	"testing"
+)
+
+// TestReplaceAttribute validates if all attributes that were replaced are present in the slog.Logger output.
+func TestReplaceAttributes(t *testing.T) {
+	var buf bytes.Buffer
+	h := slog.NewJSONHandler(&buf, &slog.HandlerOptions{
+		AddSource:   true,
+		Level:       slog.LevelDebug,
+		ReplaceAttr: replaceSlogAttributes,
+	})
+
+	l := slog.New(h)
+
+	l.Info("test")
+
+	var m map[string]interface{}
+	err := json.Unmarshal(buf.Bytes(), &m)
+	if err != nil {
+		t.Fatalf("unexpected error: %v", err)
+	}
```

Comment:
Could we use `require.NoError` instead? We're currently trying our best to decrease usage of t.Fatal 😬